File: api-net/Models/OtpCode.cs
namespace LiftOtonom.Api.Models;

public class OtpCode
{
    public long Id { get; set; }
    public long? TenantId { get; set; }
    public string Phone { get; set; } = "";
    public string Code { get; set; } = "";
    public string Purpose { get; set; } = "login";
    public short Attempts { get; set; }
    public DateTime ExpiresAt { get; set; }
    public DateTime? ConsumedAt { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: api/app/Models/OtpCode.php
<?php
// app/Models/OtpCode.php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class OtpCode extends Model
{
    public $timestamps = false;

    protected $fillable = [
        'tenant_id', 'phone', 'code', 'purpose',
        'attempts', 'expires_at', 'consumed_at',
    ];

    protected function casts(): array
    {
        return [
            'expires_at'  => 'datetime',
            'consumed_at' => 'datetime',
            'attempts'    => 'integer',
        ];
    }

    public function isExpired(): bool
    {
        return $this->expires_at->isPast();
    }

    public function isConsumed(): bool
    {
        return $this->consumed_at !== null;
    }
}
